### ISSUE TITLE
Loss function is not computed.

### ISSUE THREAD

**Issue by luforestal** [weecology/DeepForest#335]
Hello,

I am trying to train with RGB images of different size. However, since epoch 0 the loss function is not being computed. I only have a NaN from the middle of the iterations at best. 

```
annotations_file = ('train.csv')
validation_file =  ('val.csv')
m = main.deepforest()
callback = ModelCheckpoint(dirpath='temp/dir',
                                 monitor='box_recall', 
                                 mode="max",
                                 save_top_k=1,
                                 filename="box_recall-{epoch:02d}-{box_recall:.2f}")

m.config['gpus'] = '0'
m.config["train"]["csv_file"] = annotations_file
m.config["train"]["root_dir"] = 'train'
m.config["score_thresh"] = 0.4
m.config["train"]['epochs'] =  1
m.config["validation"]["csv_file"] = validation_file
m.config["validation"]["root_dir"] = 'val'
m.create_trainer(logger=TensorBoardLogger(save_dir='logdir/'), 
                                  callbacks=[callback])

m.use_release()
start_time = time.time()
m.trainer.fit(m)

save_dir = os.path.join(os.getcwd(),'pred_result')
try:
    os.mkdir(save_dir)
except FileExistsError:
    pass
results = m.evaluate(annotations_file, annotations_file, iou_threshold = 0.4, savedir= save_dir)
print(f"--- Training on CPU: {(time.time() - start_time):.2f} seconds ---")
```

![image](https://user-images.githubusercontent.com/76424439/185345709-9b7a055c-b6df-4811-855c-f2a7c664101b.png)

 - Python version and environment : Python 3.8.10 

Also, I have not been able to save any good checkpoints:

m.trainer.save_checkpoint('last_checkpoint.pl')

**Comment by bw4sz** [comment]
Just as a sanity check, can you share your annotations file, I might have a malformed box or class. I'd be interested in exploring this more, retinanet should be formatting the sizes to a standard on entry, and I have an open issue https://github.com/weecology/DeepForest/issues/347 to allow users more control over this, but in principle, torchvision should allow you to batch images of different sizes. Can you confirm whether it works with images of the same size? What use case is this where there are images of many sizes?

**Comment by ethanwhite** [comment]
Sorry we lost track of this @luforestal. If you're still looking for help can you confirm that this issue is still happening on the current release and then we can take another look.
